workflow/template_validator.go

```diff
@@ -60,36 +60,48 @@ func MustParseFromFile(path string) *Workflow {
 	return MustParse(content)
 }
 
-// RenderTemplate renders the workflow template wrt given hardware details.
+// RenderTemplate renders the workflow template with regard to the given hardware details.
 func RenderTemplate(templateID, templateData string, devices []byte) (string, error) {
-	var hardware map[string]interface{}
+	var hardware map[string]string
 	err := json.Unmarshal(devices, &hardware)
 	if err != nil {
 		err = errors.Wrapf(err, errTemplateParsing, templateID)
 		return "", err
 	}
 
+	_, buf, err := RenderTemplateHardware(templateID, templateData, hardware)
+	if err != nil {
+		return "", err
+	}
+	return buf.String(), nil
+}
+
+// RenderTemplateHardware renderst the workflow template and returns the Workflow andzzzz the interpolated bytes.
```

Comment:
andzzzz?

Reply:
lol holding z is my layer change key on my Ergodox